src/StringIntersection.java

```diff
@@ -36,6 +37,69 @@ public static String intersection(String str1, String str2) {
         return output;
 	}
 	
+	
+	
+	
+	/**
+	 * method that takes two strings and returns the intersection,
+	 * which is the string of characters that exist in both strings
+	 * order is not preserved
+	 * 
+	 * This is accomplished with a hash map.
+	 * HashMap<Character, Integer>
+	 * first put a count of all chars in str1 into HashMap
+	 * loop thru str2 and if get(c) > 0 
+	 * then output.append and replace(Int--)
+	 *  
+	 */
+	public static String intersectionHM(String str1, String str2) {
+
+
+		// create hashmap to hold Character => Integer 
+		HashMap<Character, Integer> hm = new HashMap<Character, Integer>();
+
+		Character c;
+		String output = "";
+
+		for(int i = 0; i < str1.length(); i++) {
+			c = new Character(str1.charAt(i));
+			/*
+			 * loop through str1 to create or increment the
+			 * Integer value for the count of chars c  in str1
+			 * if a char exists in str1 one or more times,
+			 * the Integer value will be the count of chars
+			 * 
+			 */
+			if(hm.containsKey(c))
+				hm.replace(c, new Integer(hm.get(c).intValue() + 1));
+			else
+				hm.put(c, new Integer(1));
+		}
+
+		// do I need to use a different name for the int iterator
```

Comment:
no, because 'i' is declared in the context of the for loop, it is garbage collected when the for loop terminates. remove comment.